src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2642,6 +2726,21 @@ export class StudyViewPageStore
         onError: () => false,
         default: false,
     });
+    @action.bound
+    updateCustomDataIntervalFilters(
+        chartUniqueKey: string,
+        dataBins: DataBin[]
+    ): void {
+        const values: DataFilterValue[] = getDataIntervalFilterValues(dataBins);
+        this.setCustomChartNumericalFilters(chartUniqueKey, values);
+        if (!this._customDataBinFilterSet.has(chartUniqueKey)) {
```

Comment:
seems strange that we update two different things here.   lets discuss.   at very least lets put some comments explaining why two different things need to be set.

Reply:
This now is implemented in  `getCustomDataNumerical()`. When the user uploads a custom data set, we don't know the data bin values beforehand (unlike a custom categorical data where we know the categories). After obtaining the binned data from the API, we need to set the bins in the filter.